azext_edge/edge/providers/support/base.py

```diff
@@ -363,6 +363,70 @@ def process_config_maps(
     )
 
 
+def process_cluster_roles(
+    directory_path: str,
+    label_selector: Optional[str] = None,
+    field_selector: Optional[str] = None,
+) -> Dict[str, Union[dict, str]]:
+    from kubernetes.client.models import V1ClusterRoleList
+
+    processed = []
+    rbac_api = client.RbacAuthorizationV1Api()
+
+    cluster_roles: V1ClusterRoleList = rbac_api.list_cluster_role(
+        label_selector=label_selector, field_selector=field_selector
+    )
+    for role in cluster_roles.items:
+        namespace = (
+            role.metadata.annotations.get("meta.helm.sh/release-namespace") if role.metadata.annotations else None
+        )
+        name = role.metadata.name
+        resource_type = _get_resource_type_prefix(BundleResourceKind.clusterrole.value)
+
+        if namespace:
+            processed.append(
+                {
+                    "data": generic.sanitize_for_serialization(obj=role),
+                    "zinfo": f"{namespace}/{directory_path}/{resource_type}.{name}.yaml",
+                }
+            )
+
+    return processed
+
+
+def process_cluster_role_bindings(
+    directory_path: str,
+    label_selector: Optional[str] = None,
+    field_selector: Optional[str] = None,
+) -> Dict[str, Union[dict, str]]:
+    from kubernetes.client.models import V1ClusterRoleBindingList
+
+    processed = []
+    rbac_api = client.RbacAuthorizationV1Api()
+
+    cluster_role_bindings: V1ClusterRoleBindingList = rbac_api.list_cluster_role_binding(
+        label_selector=label_selector, field_selector=field_selector
+    )
+    for binding in cluster_role_bindings.items:
+        namespace = (
+            binding.metadata.annotations.get("meta.helm.sh/release-namespace")
+            if binding.metadata.annotations
```

Comment:
Given the conditional assumes `binding.metadata.annotations` always exists, is it not always a `dict`? It can be `None`?

Reply:
if binding.metadata.annotations exists, it will be a dict but might not have a key "meta.helm.sh/release-namespace"?

Reply:
As the code is written, the if condition assumes `binding.metadata.annotations` always exists, and evaluates if its truthy or not. 

If this is the case, that `binding.metadata.annotations` **always** exists and is **always** a dict then this could be simplified to 

```
namespace = binding.metadata.annotations.get("meta.helm.sh/release-namespace")
```

If `bindings.metadata.annotations` may not always exist the logic can be simplified to something like

```
namespace = getattr(bindings.metadata, "annotations", None) and bindings.metadata.annotations.get("meta.helm.sh/release-namespace")
```

Reply:
from what I see, annotation might not always exist, thanks will update!